Make waveform non-experimental and first order the default (#1684)
precice/precice: docs/changelog/1684.md

> * Use `waveform-order="1"` as default. Makes time interpolation non-experimental, therefore no need to use `experimental="true"`. Note on explicit schemes: Depending on the scheme for some participants `waveform-order="1"` technically leads to a zeroth order interpolation.

Reviewer: ```suggestion
* Changed `waveform-order="1"` to be default and time interpolation to non-experimental (no need to use `experimental="true"` anymore). Note on explicit schemes: Depending on the scheme for some participants `waveform-order="1"` technically leads to a zeroth order interpolation.
```
Explicit schemes probably needs some revisiting. Maybe, we need some error messages there.
Not for this PR, but for v3: Why not cubic as default?

Author: > Explicit schemes probably needs some revisiting.

The current situation is not ideal, but it is also not easy to make the connection from `read-data` to the scheme being used and then raise the right error. Any idea for a situation where we want to raise an error? Generally, it is not wrong to ask for higher-order when using an explicit scheme. The user might just not get anything with higher order, because there is just no higher-order data available.

>  Not for this PR, but for v3: Why not cubic as default?

For cubic interpolation we need to do at least 3 substeps per window. Is see two competing arguments here:

* From the user perspective it might be surprising, if we say "the default is third order" and then third order is not reached for the setting that most users will probably run (no subcycling).
* Using a higher-order interpolation by default is the "smartest" option, because it would automatically use a higher-order interpolation, if possible. Otherwise, it would use a lower-order interpolation. Using third order is an arbitrary, but common option (see, e.g. [interpolation in scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.make_interp_spline.html#scipy.interpolate.make_interp_spline))

Main usability question here is: simple and easy to understand behavior of preCICE vs. a smart strategy that might need some more explanation. I tend towards using third order as default, because this is also what I think most libraries often do, if there is different algorithms are available ([scipy.optimize.minimize](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize) applies some "magic" for picking the best `method`, if the user does not explicitly ask for a specific method).

Reviewer: I see the doubts with cubic as default. Let's stick with "1" for the moment. That's the safe bet.

Reviewer: Explicit coupling: don't you have everything needed in the code below?
I would do warnings, not errors. "You picked order 3, but we can only do order 0".

Reviewer: > But if our default is order 1, then we would also have to raise a warning for our default value. 

Could still prevent users form wrong choices, no?

Author: I created https://github.com/precice/precice/issues/1690 to conclude the discussion about the default order for this PR. Third order is not available yet, so let's use 1st order for now.

W.r.t error messages I will also develop a solution based on 1st order interpolation as default.

Author: I decided to remove the usability checks for explicit coupling schemes. Main reason: If we also think about compositional coupling schemes, then we might actually run into contradictions, if we trigger a warning based on the `waveform-order` provided to `read-data`. And: Even for an explicit coupling scheme it is not wrong to ask for a higher waveform order (you might just not get it) @uekerman: We can also continue the discussion in an issue, if you want.
